packages/trader/src/Stores/Modules/Trading/trade-store.ts

```diff
@@ -1451,12 +1491,48 @@ export default class TradeStore extends BaseStore {
                     });
                 }
             }
+            if (this.is_turbos && response.error.details?.payout_per_point_choices) {
+                const { payout_per_point_choices, min_stake, max_stake } = response.error.details;
+                const payoutIndex = Math.floor(payout_per_point_choices.length / 2);
+                this.setPayoutChoices(payout_per_point_choices);
+                this.setStakeBoundary(contract_type, min_stake, max_stake);
+                this.onChange({
+                    target: {
+                        name: 'payout_per_point',
+                        value: payout_per_point_choices[payoutIndex],
+                    },
+                });
+
+                this.barrier_1 = String(
+                    (Number(response.proposal?.contract_details?.barrier) - Number(response.proposal?.spot)).toFixed(3)
+                );
+            }
         } else {
             this.validateAllProperties();
-            if (this.is_turbos || this.is_vanilla) {
+            if (this.is_vanilla) {
                 const { max_stake, min_stake, barrier_choices } = response.proposal ?? {};
                 this.setBarrierChoices(barrier_choices as string[]);
                 this.setStakeBoundary(contract_type, min_stake, max_stake);
+            } else if (this.is_turbos) {
+                const { max_stake, min_stake, payout_choices } = response.proposal ?? {};
+                if (payout_choices) {
+                    this.setPayoutChoices(payout_choices as number[]);
+                    this.setStakeBoundary(contract_type, min_stake, max_stake);
+                    if (
+                        this.barrier_1 !==
+                        String(
+                            (
+                                Number(response.proposal?.contract_details?.barrier) - Number(response.proposal?.spot)
+                            ).toFixed(3)
```

Comment:
Shouldn't the decimal places be based on the symbol's decimal place?

Reply:
Setting the fixed decimal with barrier pipsize now as a temporary fix until we get the value from the BE.